packages/database/src/core/SyncTree.ts

```diff
@@ -1049,3 +998,43 @@ function syncTreeSetupListener_(
   }
   return events;
 }
+
+/**
+ * Added to support non-listener data queries (Used by `get()`).
+ * Simulates `onValue` `onlyOnce` where it adds the event registration,
+ * and then applies the changes to the tree.
+ * @param repo - repo reference to the overall Repo.
+ * @param query - query to be added.
+ * @param node - Data needed to be added to the tree
+ * @param eventRegistration - a dummy event registration to add to the SyncTree
+ */
+export function syncTreeAddGetRegistration(
```

Comment:
I think this function is too similarly named to syncTreeAddEventRegistration, which only adds an event registration whereas this one also updates data in the synctree itself. 

IMO we should either rename this to `syncTreeRegisterGetResults` or inline it in `repoGetValue`.

Reply:
Inlined instead.